screens/feed/FeedScreen.tsx

```diff
@@ -1,76 +1,93 @@
+import { FeedData, FeedCategory, ActivityFeedData, AchievementFeedData } from '../../types/feedTypes';
+import { useFeedQuery } from '../../graphql/queries/Feed.generated';
+import { FlatList } from 'react-native-gesture-handler';
+import Loading from '../../components/general/Loading';
+import Error from '../../components/general/Error';
+import { convertToFeedData } from '../../helpers/objectMappers';
+
+const getItem = (data: FeedData) => {
+  if (data.feedCategory === FeedCategory.ACTIVITY) {
+    return (
+      <View style={styles.element}>
+        <ActivityFeedCard data={data as ActivityFeedData} />
+      </View>
+    );
+  } else if (data.feedCategory === FeedCategory.ACHIEVEMENT) {
+    return (
+      <View style={styles.element}>
+        <AchievementFeedCard data={data as AchievementFeedData} />
+      </View>
+    );
+  }
+  return <></>;
+};
 
 const FeedScreen: React.FC = () => {
-  const testGeoFence: CircleGeoFence = {
-    id: 9,
-    latitude: 63.431731,
-    longitude: 10.406626,
-    radius: 50,
-    name: 'Test name',
-    category: GeoFenceCategory.EDUCATION,
-    variant: GeoFenceVariant.CIRCLE,
-    description: 'Test description',
-  };
-  const testActivity: ActivityFeedData = {
-    userName: 'Siri Mykland',
-    caption: 'Very nice activity!',
-    geoFence: testGeoFence,
-    startedAt: '2021-01-29T10:00:00+01:00',
-    picture: 'https://api.multiavatar.com/Kuninori%20Bun%20Lord.png',
-    score: 200,
-    duration: 100,
-  };
-  const testAchievement: AchievementFeedData = {
-    userName: 'Mats Tyldum',
-    picture: 'https://api.multiavatar.com/c68e82f2fb46979b85.png',
-    achievement: {
-      name: '1000 points',
-      description: 'Achieved 1000 points!',
-      type: AchievementVariant.SCORE,
-      createdAt: '2021-01-28T09:00:00+01:00',
-      level: 1,
-      rule: { score: 100, category: 'CULTURE' },
+  const [refreshing, setRefreshing] = useState(false);
+  const [limit, setLimit] = useState(10);
+  const [feedElements, setFeedElements] = useState<FeedData[]>([]);
+
+  const { loading: loading, error: error, data: data, refetch, fetchMore } = useFeedQuery({
+    variables: {
+      limit: limit,
     },
+    nextFetchPolicy: 'network-only',
+  });
+  useEffect(() => {
+    if (data && data.feed) {
+      const feedData = convertToFeedData(data);
+      setFeedElements(feedData);
+    }
+  }, [data]);
+
+  const onRefresh = async () => {
+    setRefreshing(true);
+    await refetch();
+    setRefreshing(false);
   };
-  const testChallenge: ChallengeFeedData = {
-    name: 'MaxPoints7Days',
-    userName1: 'Mats Tyldum',
-    userName2: 'Siri Mykland',
-    userPicture1: 'https://api.multiavatar.com/c68e82f2fb46979b85.png',
-    userPicture2: 'https://api.multiavatar.com/Kuninori%20Bun%20Lord.png',
-    createdAt: '2021-01-29T13:00:00+01:00',
-    description: 'Highest amount of points in the next 7 days.',
+  const loadMoreFeedElements = async () => {
+    const newLimit = limit + 10;
```

Comment:
With this query, you are fetching the same objects again and again but 10 more each time, should consider using both limit and offset. Then we can append them to the feedElements.